### PR TITLE
Master payroll auto loading lock conflict sed

### PR DESCRIPTION
**Pull request by sed-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_payroll/models/hr_benefit.py [hunk 1]
```diff
@@ -279,11 +279,13 @@ def _duplicate_to_calendar_attendance(self):
     def action_validate(self, ids):
         benefits = self.env['hr.benefit'].search([('id', 'in', ids), ('state', '!=', 'validated')])
         benefits.write({'display_warning': False})
-        if not benefits._check_if_error():
-            benefits.write({'state': 'validated'})
-            benefits._duplicate_to_calendar()
-            return True
-        return False
+
+        no_conflict_benefits = benefits.filtered(lambda x: not x._check_if_error())
```
Comment (tivisse) [inline addons/hr_payroll/models/hr_benefit.py]: No break the batch method. 

Maybe modify ``_check_if_error()`` to return (bool, benefits_in_error) and avoid to use a filtered.

File: addons/hr_payroll/models/hr_employee.py [hunk 2]
```diff
@@ -100,3 +100,4 @@ def _format_datetime(date):
 
         new_benefits = self.env['hr.benefit']._safe_duplicate_create(vals_list, date_start, date_stop)
         new_benefits._compute_conflicts_leaves_to_approve()
+        new_benefits.action_validate(new_benefits.ids)
```
Comment (tivisse) [inline addons/hr_payroll/models/hr_employee.py]: Let's modify ``action_validate`` to use recordsets.

File: addons/hr_payroll/static/src/js/benefit.js [hunk 3]
```diff
@@ -12,9 +12,84 @@ odoo.define('hr_payroll.benefit.view_custo', function(require) {
...
+            'click .btn-benefits-conflict': '_onBenefitsConflict',
+        },
+
+        /**
+         * @override
+         */
+        reload: function () {
+            var date_fmt = 'YYYY-MM-DD HH:mm:ss';
+            var self = this;
+            this.firstDay = this.model.data.target_date.clone().startOf('month');
```
Comment (tivisse) [inline addons/hr_payroll/static/src/js/benefit.js]: Seem louche, to check with someone smarter than me.

File: addons/hr_payroll/static/src/js/benefit.js [hunk 4]
```diff
@@ -12,9 +12,84 @@ odoo.define('hr_payroll.benefit.view_custo', function(require) {
...
+            var super_function = this._super;
+            var args = arguments;
+
+            if (this.firstDay < this.end_current_month)
+            {
+                return this._rpc({
+                    model: 'hr.employee',
+                    method: 'generate_benefit',
+                    args: [this.firstDay.format(date_fmt), this.lastDay.format(date_fmt)],
+                }).then(function () {
```
Comment (tivisse) [inline addons/hr_payroll/static/src/js/benefit.js]: Rebase and adapt to new Promise mechanism.

File: addons/hr_payroll/static/src/js/benefit.js [hunk 5]
```diff
@@ -12,9 +12,84 @@ odoo.define('hr_payroll.benefit.view_custo', function(require) {
...
+            else
+            {
+                return super_function.apply(self, args);
+            }
+        },
+
+        /**
+         * @override
+         */
+        start: function () {
```
Comment (tivisse) [inline addons/hr_payroll/static/src/js/benefit.js]: Try to extract the date manipulation + rpc code to a specific method and reuse it in the ``start`` and the ``reload``

File: addons/hr_payroll/models/hr_benefit.py [hunk 6]
```diff
@@ -81,11 +81,11 @@ def _check_if_error(self):
         undefined_type.write({'display_warning': True})
         conflict = self._mark_conflicting_benefits(min(self.mapped('date_start')), max(self.mapped('date_stop')))
         conflict_with_leaves = self._compute_conflicts_leaves_to_approve()
-        return undefined_type or conflict or conflict_with_leaves
+        return undefined_type + conflict + conflict_with_leaves
```
Comment (LucasLefevre) [inline addons/hr_payroll/models/hr_benefit.py]: What about union instead of concatenation?

Comment (sed-odoo) [reply addons/hr_payroll/models/hr_benefit.py]: Good idea, it will be corrected in next rebase
Thanks

**Pull request closed without merging**